.github/workflows/pr.yml

```diff
@@ -1,27 +1,90 @@
 name: Build CI
 
 on:
+  push:
+    branches: [ "master" ]
   pull_request:
     branches: [ "master" ]
 
 jobs:
   build:
-
+    name: Run build workflow
     runs-on: ubuntu-latest
 
     strategy:
       matrix:
         node-version: [16.x]
-        # See supported Node.js release schedule at https://nodejs.org/en/about/releases/
 
     steps:
-    - uses: actions/checkout@v3
-    - name: Use Node.js ${{ matrix.node-version }}
-      uses: actions/setup-node@v3
-      with:
-        node-version: ${{ matrix.node-version }}
-        cache: 'npm'
-    - run: npm ci
-    - run: npm run lint
-    - run: npm run test
-    - run: npm run build
\ No newline at end of file
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v3
+        with:
+          node-version: ${{ matrix.node-version }}
+          cache: 'npm'
+
+      - name: Run CI
+        run: npm ci
+
+      - name: Run Test
+        run: npm run test
+
+      - name: Run Build
+        run: npm run build
+
+  eslint:
+    name: Run eslint code scanning
+    runs-on: ubuntu-latest
+
+    permissions:
+      security-events: write
+
+    strategy:
+      fail-fast: false
+      matrix:
+        language: [ 'javascript' ]
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Install ESLint
```

Comment:
does this eslint uses react built-in lint?
current workflow using built-in one

Reply:
ok, so we dont need to install eslint, fixed

Reply:
react's lint avalible only from build jobs where project has been installed

Reply:
actually i dont understand what you whant me to do

Reply:
Current workflow run react's built in lint here 
https://github.com/fakng-agrgtr/fakng-ui/blob/4904ee546e6bc157adfba0d3d2b9e28024684188/.github/workflows/pr.yml#L25

So there is no need at this job at all. Revert to current master's version

Reply:
I move lint check to separate job on purpose, for speed up and ability to upload eslint results to GitHub code ql, therefore it is identical to previous npm run lint and uses the same .eslint config. I don't think it's necessary to also run lint step on build phase,

Reply:
There is no explicit installation of eslint and eslint react plugin. It do resolve it from project's package.
The eslint job do not uses clean install. So its could cause some troubles.
If you need formatter install it in package and it gives us a new package-lock. Then you could upload eslint results to github even from current's pr workflow with couple changes in workflow and in package scripts.

We do not need project build before lint errors solved. That build stays unnecessary forever after fixes commitet